opencog/attention/experimental/HebbianUpdatingAgent.cc

```diff
@@ -52,7 +57,12 @@ void HebbianUpdatingAgent::run()
     std::back_insert_iterator< std::vector<Handle> > out_hi(atoms);
 
     _as->get_handle_set_in_attentional_focus(out_hi);
-
+     /*Without adding this sleep code right below the above method call,
+     * nlp-parse evaluation thread waits for minutes before it gets a chance to
+     * run.
+     */
+    std::this_thread::sleep_for(std::chrono::nanoseconds(1));
```

Comment:
OK I updated #2333 to describe the bug that you are hitting here, and to describe how to fix it.  It will require some surgery.  Its not hard, per se, but it requires surgical skills and a steady hand, to not screw things up.